### PR TITLE
Node-RED Docs:  How to implement loops in Node-RED flows

### PR DESCRIPTION
**Pull request by sumitshinde-84**
## Checklist

 - [ ] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] I have considered the performance impact of these changes
 - [ ] Suitable unit/system level tests have been added and they pass
 - [ ] Documentation has been updated

### CODE REVIEW

**Comment by Hasmin-AC** [comment]
How is this blog related to FF platform capabilities?  Is it possible to give use case and/or scenarios in the manufacturing industrial setting?

**Review by Hasmin-AC — COMMENTED** [review]
Need to relate this content to our ICP more. Give examples of how this helps in an industrial setting.

**Comment by sumitshinde-84** [comment]
@Hasmin, this is one of the most common Node-RED solutions for people who are searching frequently. I think we should keep it to Node-RED basics

**Comment by sumitshinde-84** [comment]
This is a topic that people search for a lot but don't have many resources on.

**Comment by Hasmin-AC** [comment]
Lets move it to Node-RED docs then.

**Comment by sumitshinde-84** [comment]
> Lets move it to Node-RED docs then.

Under which category?

**Comment by ZJvandeWeg** [comment]
> > Lets move it to Node-RED docs then.
> 
> Under which category?

I think we have a "Programming" section now?

**Comment by sumitshinde-84** [comment]
Hi ZJ, I have reviewed the docs and made some updates. Could you please review them?

File: src/node-red/getting-started/programming/loop.md [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+## What is a Loop?
+
+A [loop](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Loops_and_iteration) is a programming construct that allows you to execute a block of code repeatedly until a certain condition is met. Different types of loops are suited to different scenarios, depending on how and when you want the code to repeat:
+
+- **For Loop**: Executes a block of code a specific number of times. This is useful when you know in advance how many times you need to iterate, such as iterating through a range of numbers or items in a list. It's also known as a fixed number loop.
+
+- **While Loop**: Repeats a block of code as long as a specified condition is true. This type of loop is useful when you don’t know how many times the loop will need to run beforehand. The loop continues executing until the condition becomes false.
+
+- **Do...While Loop**: Similar to the while loop, but with a key difference: it guarantees that the code block will execute at least once before checking the condition. This is because the condition is evaluated after the loop’s code has executed.
+
```
Comment (ZJvandeWeg) [inline src/node-red/getting-started/programming/loop.md:26]: Do while and a while loop are the same. Feels like we should differentiate the 3 cases:
1. For Each - Iterate over a set of elements and perform a task on each of them
2. While - Perform a task until a certain exit criteria is met
3. For - Execute a certain block a certain numer of times.

**Pull request merged**